### PR TITLE
HPCC-19417 CmdGetDir duplicate directory in file path

### PR DESCRIPTION
**Pull request by mckellyln**
HPCC-18725 update

Signed-off-by: Mark Kelly <mark.kelly@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:

QA suite passed as expected
Kevin tried to download slave logs from this dafilesrv version

### CODE REVIEW

**Comment by mckellyln** [comment]
@wangkx fyi

**Comment by mckellyln** [comment]
@jakesmith please review.
Is there any other testing we should do, besides trying to get slave logs and the regression suite (which have already been done) ?

File: common/remote/sockfile.cpp [hunk 1]
```diff
@@ -6635,6 +6635,36 @@ class RemoteFileTest : public CppUnit::TestFixture
         iFile2->remove();
         iFile->move(subDirFilePath);
 
+        // open sub-directory file2 explicitly
+        RemoteFilename rfn;
+        rfn.setRemotePath(subDirPath.str());
+        SocketEndpoint ep(serverPort);
+        rfn.setIp(ep);
```
Comment (jakesmith) [inline common/remote/sockfile.cpp]: setRemotePath should really be used with a url, better to do:
```
    rfn.setPath(ep, subDirPath.str());
```

File: common/remote/sockfile.cpp [hunk 2]
```diff
@@ -6635,6 +6635,36 @@ class RemoteFileTest : public CppUnit::TestFixture
...
+
+        Owned<IFile> dir = createIFile(rfn);
+        Owned<IDirectoryIterator> diriter = dir->directoryFiles("file2");
+        if (!diriter->first())
+        {
+            fflush(NULL);
+            fprintf(stdout, "Error, file2 diriter->first() is null\n");
+            fflush(NULL);
```
Comment (jakesmith) [inline common/remote/sockfile.cpp]: Any reason this (and others below) aren't using PROGLOG ?

File: common/remote/sockfile.cpp [hunk 3]
```diff
@@ -6635,6 +6635,36 @@ class RemoteFileTest : public CppUnit::TestFixture
...
+        diriter.clear();
+        dir.clear();
+
+        OwnedIFileIO iFile3IO = iFile3->openShared(IFOread, IFSHfull);
+        if (!iFile3IO)
+        {
+            fflush(NULL);
+            fprintf(stdout, "Error, file2 openShared() failed\n");
+            fflush(NULL);
+            CPPUNIT_ASSERT(0);
```
Comment (jakesmith) [inline common/remote/sockfile.cpp]: Better to use CPPUNIT_ASSERT_MESSAGE to return meaningful message.

File: common/remote/sockfile.cpp [hunk 4]
```diff
@@ -6659,9 +6689,18 @@ class RemoteFileTest : public CppUnit::TestFixture
...
+        try
+        {
+            iFile2->setFilePermissions(0777);
+        }
+        catch (...)
+        {
+            fflush(NULL);
+            fprintf(stdout, "iFile2->setFilePermissions() exception\n");
+            fflush(NULL);
+            CPPUNIT_ASSERT(0);
```
Comment (jakesmith) [inline common/remote/sockfile.cpp]: Better to use CPPUNIT_ASSERT_MESSAGE to return meaningful message.

**Comment by jakesmith** [comment]
@mckellyln - some minor comments

**Comment by mckellyln** [comment]
@jakesmith updated

**Comment by jakesmith** [comment]
Looks good.
@richardkchapman - ready to merge.

**Pull request merged**